spring-boot-project/spring-boot/src/main/java/org/springframework/boot/json/JsonValueWriter.java

```diff
@@ -106,8 +106,8 @@ else if (value instanceof Map<?, ?> map) {
 		else if (value instanceof Number) {
 			append(value.toString());
 		}
-		else if (value instanceof Boolean) {
-			append(Boolean.TRUE.equals(value) ? "true" : "false");
+		else if (value instanceof Boolean bool) {
+			append(bool ? "true" : "false");
```

Comment:
I think it would be better to use `toString()` here. It could then be combined with the handling of `Number`.